### ISSUE TITLE
Provide an overview of supported Router implementations

### ISSUE THREAD

**Issue by AlexanderBand** [NLnetLabs/rpki-faq#3]
- Cisco
- Juniper
- Nokia/Alcatel-Lucent
- BIRD
- Quagga
- RTRLib usage
- More?

**Comment by job** [comment]
we should separate which have support for RTR, and which have support for
Origin Validation through 'static' configuration.

OpenBGPD has support for static, but not for RTR.

**Comment by job** [comment]
Cisco IOS: no support
Cisco IOS XE: support
Cisco IOS XR: support

**Comment by AlexanderBand** [comment]
Juniper JUNOS 12.2 >

**Comment by maelmans** [comment]
We have an item open with engineering for RFC8210 support. I'm trying to find out if it's already committed and if so in which Junos release. Could be interesting to add here.

**Comment by AlexanderBand** [comment]
Is this still accurate for Cisco?

**High-end & mid-range routers running IOS-XR**
Minimum release XR 4.2.1:

- CRS-1, CRS-3, CRS-x
- ASR9000
- c12000

As of XR 5.1.1:

- NCS6000 (optical router)
- XRv (virtual router on x86)

**Access/Enterprise routers running IOS-XE**
Minimum release XE 3.5:

- c7200, c7600, ASR1000
- CSR1000v (Virtual Router on x86)
- ASR901, ASR903, ASR907
- ME3600, ME3800

ASR1000 & CSR1000v also support BGP route-server functions with RPKI

**Comment by waehlisch** [comment]
I will add something about RTRlib and its integration into FRR and BIRD.

**Comment by lukastribus** [comment]
@AlexanderBand c7200, c7600, ME3600, ME3800 do not run IOS-XE, but legacy IOS. Instead you probably want to add the ASR920, which is relevant and runs IOS-XE (although it does not cope with a full table, just like other platforms on this list).

**Comment by ghankins** [comment]
Nokia (static and RTR):
- 7210 SAS
- 7750 SR
- 7950 XRS
- VSR

**Comment by maelmans** [comment]
I have this list from *somewhere*....I think it was the RIPE website?

Juniper:
- Junos OS has support since Junos OS version 12.2

Cisco: 
- XR 4.2.1 (CRS-x, ASR9000, c12K) / XR 5.1.1 (NCS6000, XRv) 
- XE 3.5 (C7200, c7600, ASR1K, CSR1Kv, ASR9k, ME3600…) 
- IOS15.2(1)S

Others:
- Alcatel Lucent has support since SR-OS 12.0 R4
- Quagga has support through BGP-SRX
- BIRD has support for ROA but does not do RPKI-RTR

I know for sure that for BIRD it's not completely actual anymore.

**Comment by AlexanderBand** [comment]
@maelmans, I recognise that list from a RIPE NCC slide set, such as [this one](https://indico.uknof.org.uk/event/35/contribution/15/material/slides/0.pdf). It's most likely based on [this list](https://www.ripe.net/manage-ips-and-asns/resource-management/certification/router-configuration#Cisco) was given to me by my Cisco contact @bduvivie at the time (~2011?). I'm basically looking for an up to date list for Cisco.

For Quagga you can use RTRLib, as per @waehlisch, but it looks like [BGP-SRX](https://www.nist.gov/services-resources/software/bgp-secure-routing-extension-bgp-srx-prototype) is also still maintained. Perhaps @dougm-nist can shed some light on that topic.

**Comment by dougm-nist** [comment]
Yes BGP-SrX is still developed/maintained.    Most of our recent work focuses on BGPsec, but the platform is still maintained.    Actually in the next few months we will release other ports of our crypto API and proxy interface for GoBGP and ExaBGP.     https://www.nist.gov/services-resources/software/bgp-secure-routing-extension-bgp-srx-prototype

**Comment by dougm-nist** [comment]
While I am at it, you should add GoBGP to your list.

https://github.com/osrg/gobgp

**Comment by AlexanderBand** [comment]
We should also add [GoRTR](https://github.com/cloudflare/gortr) by @lspgn.

**Comment by dougm-nist** [comment]
GoRTR is a RPKI-to-RTR proxy as far as I know ... not an implementation of BGP-OV.  So it would go in a list of RPKI validation components, not router implementations.

**Comment by AlexanderBand** [comment]
Another list found [here](http://www.bgp4all.com.au/dokuwiki/_media/workshops/02-rpki.pdf):
- Cisco IOS – available in release 15.2
- Cisco IOS/XR – available in release 4.3.2
- Juniper – available in release 12.2
- Nokia – available in release R12.0R4
- Huawei – newly available – release TBA
